sui_programmability/examples/games/tests/randomness_based_lottery_tests.move

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[test_only]
+module games::randomness_based_lottery_tests {
+    use sui::test_scenario::{Self, Scenario};
+    use games::randomness_based_lottery::{Self, Lottery, Ticket};
+    use sui::sui::SUI;
+    use sui::coin;
+    use sui::transfer;
+    use sui::coin::Coin;
+    use sui::randomness::Randomness;
+    use std::option;
+    use sui::randomness;
+
+    fun mint(addr: address, amount: u64, scenario: &mut Scenario) {
+        transfer::transfer(coin::mint_for_testing<SUI>(amount, test_scenario::ctx(scenario)), addr);
+        test_scenario::next_tx(scenario, addr);
+    }
+
+    #[test]
+    fun test_play_randomness_lottery() {
+        let user1 = @0x0;
+        let user2 = @0x1;
+        let user3 = @0x2;
+        let user4 = @0x3;
+
+        let scenario_val = test_scenario::begin(user1);
+        let scenario = &mut scenario_val;
+
+        randomness_based_lottery::create(test_scenario::ctx(scenario));
+        test_scenario::next_tx(scenario, user1);
+        let lottery = test_scenario::take_shared<Lottery>(scenario);
+
+        // User1 buys a ticket.
+        test_scenario::next_tx(scenario, user1);
+        mint(user1, 1, scenario);
+        let coin = test_scenario::take_from_sender<Coin<SUI>>(scenario);
+        test_scenario::next_tx(scenario, user1);
```

Comment:
Minor: it would help to separate steps by an empty line or a parenthesis - purely for readability.